jujugui/static/gui/src/app/app.js

```diff
@@ -1100,9 +1071,7 @@ YUI.add('juju-gui', function(Y) {
           pluralize={utils.pluralize.bind(this)}
           staticURL={window.juju_config.staticURL}
           storeUser={this.storeUser.bind(this)}
-          switchModel={utils.switchModel.bind(this,
-            this.createSocketURL.bind(this, this.get('socketTemplate')),
-            this.switchEnv.bind(this), this.env)}
+          switchModel={utils.switchModel.bind(this, this.env)}
```

Comment:
Yeah switchModel uses methods on the app context. We could (and possibly should) pass those in, but the code wouldn't get any cleaner, it would simply change the bind from `this` to `utils`